src/ripple/rpc/impl/Handler.cpp

```diff
@@ -174,14 +189,39 @@ Handler const handlerArray[]{
 class HandlerTable
 {
 private:
+    using handler_table_t = std::multimap<std::string, Handler>;
+
+    // Use with equal_range to enforce that API range of a newly added handler
+    // does not overlap with API range of an existing handler with same name
+    bool
+    overlappingApiVersion(
+        std::pair<handler_table_t::iterator, handler_table_t::iterator> range,
+        unsigned minVer,
+        unsigned maxVer)
+    {
+        assert(minVer <= maxVer);
+        assert(maxVer <= RPC::apiMaximumValidVersion);
+
+        for (; range.first != range.second; range.first++)
+        {
+            if (range.first->second.minApiVer_ <= maxVer &&
+                range.first->second.maxApiVer_ >= minVer)
+                return true;
+        }
+        return false;
+    }
+
     template <std::size_t N>
     explicit HandlerTable(const Handler (&entries)[N])
     {
-        for (std::size_t i = 0; i < N; ++i)
+        for (auto const& entry : entries)
         {
-            auto const& entry = entries[i];
-            assert(table_.find(entry.name_) == table_.end());
-            table_[entry.name_] = entry;
+            assert(!overlappingApiVersion(
```

Comment:
Instead of the `assert`, can we throw a `LogicError` instead? I'm not sure if asserts are enabled in all types of builds. If there are duplicate/overlapping handlers, it sounds like we need to throw a halting error message.

Reply:
sure, can do. Note that debug binary is tested in CI (meaning, an assert would be caught), and the population of handlers is exactly the same in both debug and release.